libs/pong/pong2.js

```diff
@@ -0,0 +1,119 @@
+    const anonymousIp = anonymousIpByCC(countryCode);
+
+    const placements = pongs
+      .filter((p) => p in zoneKeys)
+      .map((p) => {
+        return { name: p, zoneKey: [zoneKeys[p]] };
+      });
+
+    const requests = placements.map(async ({ name, zoneKey }) => {
+      const {
+        ads: [
+          {
+            description = null,
+            statlink,
+            statimp,
+            smallImage,
+            backgroundColor,
+            backgroundHoverColor,
+            callToAction,
+            ctaBackgroundColor,
+            ctaBackgroundHoverColor,
+            ctaTextColor,
+            ctaTextColorHover,
+            textColor,
+            textColorHover,
+          },
+        ] = [],
+      } = await (
+        await fetch(
+          `https://srv.buysellads.com/ads/${zoneKey}.json?forwardedip=${encodeURIComponent(
+            anonymousIp
+          )}${userAgent ? `&useragent=${encodeURIComponent(userAgent)}` : ""}`
+        )
+      ).json();
+      return {
+        name,
+        p:
+          description === null
+            ? null
+            : {
+                click: coder.encodeAndSign(statlink),
+                view: coder.encodeAndSign(statimp),
+                image: coder.encodeAndSign(smallImage),
+                copy: description && he.decode(copy),
+                cta: callToAction && he.decode(callToAction),
+                colors: {
+                  textColor,
+                  backgroundColor,
+                  ctaTextColor,
+                  ctaBackgroundColor,
+                  textColorDark: textColorHover,
+                  backgroundColorDark: backgroundHoverColor,
+                  ctaTextColorDark: ctaTextColorHover,
+                  ctaBackgroundColorDark: ctaBackgroundHoverColor,
+                },
+              },
+      };
+    });
+    const decisionRes = await Promise.all(requests);
+
+    const decisions = Object.fromEntries(
+      decisionRes.map(({ name, p }) => [name, p])
+    );
+
+    if (pongs.every((p) => decisions[p] === null)) {
+      let status = "geo_unsupported";
+      return { statusCode: 200, payload: { status } };
+    }
+    const payload = Object.fromEntries(
+      Object.entries(decisions)
+        .map(([p, v]) => {
+          if (v === null) {
+            return [p, null];
+          }
+          const { copy, image, click, view, cta, colors = {} } = v;
+          return [
+            p,
+            {
+              status: "success",
+              copy,
+              image,
+              cta,
+              colors,
+              click,
+              view,
+            },
+          ];
+        })
+        .filter(Boolean)
+    );
+    return { statusCode: 200, payload };
+  };
+}
+
+export function createPong2ClickHandler(coder) {
+  return async (params) => {
+    const click = coder.decodeAndVerify(params.get("code"));
+    const res = await fetch(`https:${click}`, { redirect: "follow" });
```

Comment:
I think it might be best to catch a `null` `click` value, if verification fails, so we don't attempt to fetch `https:null`